addons/binding/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/internal/handler/HarmonyHubHandler.java

```diff
@@ -83,54 +86,96 @@
     private List<HubStatusListener> listeners = new CopyOnWriteArrayList<>();
     private final HarmonyHubHandlerFactory factory;
     private @NonNullByDefault({}) HarmonyHubConfig config;
-    private @Nullable HarmonyClient client;
+    private @NonNullByDefault({}) HarmonyClient client;
```

Comment:
Since the client is assigned only once in the constructor, I think you can simplify the code by assigning it here instead and making it final. Then you can remove the `@NonNullByDefault({})` and also all `client != null` checks. :-)

Reply:
You are correct, i blindly copied the pattern without thinking about.